### DISCUSSION TITLE
[Confusion] Multiple ways of setting page title and META tags

### DISCUSSION THREAD

**Discussion by xanderim** [nuxt/nuxt#17360] [Questions] [answered]
I am a bit confused as there are several ways to do the same thing such as adding title/ META tags. Let me explain.
You can set a title and META tags with:
- nuxt.config
- useHead composable and  titleTemplate 
- there is also `definePageMeta` which populates `route.meta` and then can be used in `useHead` composable
- useMeta (deprecated)
- and finally, there are Meta components.

Please help me to understand:
1. A good  use case for `definePageMeta`
2. Cases when `titleTemplate` is a preferred way instead of `<Title>` Meta component.
3. Can I use `useHead` to dynamically set data received from API?

**Comment by Flanker72** [comment, marked as answer]
Seems the decision is up to you.

In my opinion the answer may be like this:
- nuxt.config sets the global behavior for all pages.
- definePageMeta is a syntax sugar, and allows you to make your template more clear, moving the rendering logic to the script part.
- titleTemplate may be used with some complex logic and also makes the template more clear. Also it may be overriden on the per-page basis.
- yes, you can use `useHead` after the `await useAsyncData` to set the title, it works.
